src/crafting.cpp

```diff
@@ -2177,6 +2201,10 @@ void player::complete_disassemble( item_location &target, const recipe &dis )
                 newit.ammo_set( newit.ammo_default(), newit.ammo_capacity() );
             }
 
+            if( dis_item.is_ammo() ) { //If ammo, overwrite component count with selected quantity of ammo
```

Comment:
This should be more several lines above. Specifically somewhere before the "Compress liquids and counted-by-charges items into one item," comment. Currently this will create several *identical* entries within `components`.

I also suggest instead of simply overriding the value, to just multiply it. See https://github.com/CleverRaven/Cataclysm-DDA/pull/37709/files#diff-244fef9df19695eb07ac24b399d34e91L2152-R2176 where it currently applies the charges from the product.

Reply:
Fixed this and the redundant parenthesis